### PR TITLE
Bug 2053168: Fix dynamic plugin SDK typescript IDE errors

### PR DESCRIPTION
**Pull request by florkbr**
With some of the console logic migration into the SDK, we mistakenly used
aliased path imports. These are ignored by typescript.
IE: `@console/dynamic-plugin-sdk/src/utils/k8s/hooks`

These should be relative to play nice with typescript
IE: `../utils/k8s/hooks`

See: https://bugzilla.redhat.com/show_bug.cgi?id=2053168

### CODE REVIEW

**Review by andrewballantyne — COMMENTED** [review]
Looking good from a fixing of the problem... I wonder if we want to vet anything we have moved recently over. If you look into the `src/index.ts` file it exports a lot of things we have moved. Anything that is here now might be something we want to expose. 

`src/index` is only for the monorepo -- which is why our migration has worked well... but now that we are creating npm packages out of this effort, I wonder if we have omitted anything 🤔 

cc @fhlavac

**Comment by vojtechszocs** [comment]
> `src/index` is only for the monorepo -- which is why our migration has worked well... but now that we are creating npm packages out of this effort, I wonder if we have omitted anything thinking

@andrewballantyne `src/index` is the Console dynamic plugin SDK monorepo package main module used by other Console monorepo packages, while `src/lib-xx` are entry points for `tsc` compilation when building specific distributable SDK packages (core, webpack, internal, etc).

I'm not sure we need any kind of parity between these two kinds of entry files; they serve different purposes and have different consumers.

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ import {
 } from '../extensions/console-types';
 import { StatusPopupSectionProps, StatusPopupItemProps } from '../extensions/dashboard-types';
 
+export * from '../app/components';
+export * from './common-types';
```
Comment (florkbr) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:26]: @andrewballantyne from my investigation:

Both the monorepo entry point and lib-core entry point have:   
1. K8s utils (from dynamic-core-api.ts)
1. K8s type definitions
1. K8s hooks
1. useResolvedExtensions util
1. Components (missing - added)
1. Common-types (missing - added)
1. Extensions
1. useActivePerspective (through direct export in monorepo entrypoint and `core-api` in lib-core entry point)
1. perspective-context (missing - added; not required for core SDK so added to core-api)

Only in the monorepo entry point:
1. Appinitsdk
1. Reducers
1. Actions

I'll continue looking tomorrow. Let me know if anything jumps out at you from that list.

Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:26]: Note - somehow `AppInitSDK` is actually being pulled into the core dist directory. Need to investigate why that is happening:

image

Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:26]: Yeah, @vojtechszocs might need to look at this. I'll try to poke around and see if I can gleam what's up here.

Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:26]: FWIW, it appears to be a tree shaking issue. There are imports to the index file where it is exported and thus the bundler is just grabbing everything up. It is likely part-of but not exported-out-of the Core package.

**Review by andrewballantyne — APPROVED** [review]

**Comment by andrewballantyne** [comment]
/hold cancel

Confusion around direction/solution has been rectified removing hold.

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts (hunk 1 above)
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:26]: Fixed with https://github.com/openshift/console/pull/11034/commits/d158adea1bdf2b62defb3333ad8e7d410d68485b. For now made the imports more specific, but we will need to investigate this in the future (very easy to pull it back in accidentally).

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts [hunk 2]
```diff
@@ -1 +1,7 @@
+/**
+ * @file internal-api entrypoint for the dynamic plugin SDK. Internal package exposing additional code.
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts:2]:
```suggestion
 * @file Entrypoint for the `@openshift-console/dynamic-plugin-sdk-internal` package published to npmjs.
```

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts [hunk 3]
```diff
@@ -1 +1,7 @@
+/**
+ * @file webpack entrypoint for the dynamic plugin SDK. Provides webpack plugin ConsoleRemotePlugin used to build all dynamic plugin assets.
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts:2]:
```suggestion
 * @file Entrypoint for the `@openshift-console/dynamic-plugin-sdk-webpack` package published to npmjs.
```

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-host-app.ts [hunk 4]
```diff
@@ -1 +1,7 @@
+/**
+ * @file host-app entrypoint for the dynamic plugin SDK. Provides APIs, Component, and utilities for host applications i.e reducers, actions, etc
+ *
+ * Published to NPM
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/lib-host-app.ts:4]: This package isn't actually published to npmjs :) we can remove it in a follow-up PR.

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts [hunk 5]
```diff
@@ -1 +1,7 @@
...
+ *
+ * Published to NPM
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts:4]: This is in the first line of the doc ❓

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts [hunk 6]
```diff
@@ -1 +1,7 @@
...
+ *
+ * Published to NPM
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts:4]: Still published to NPM though

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-host-app.ts (hunk 4 above)
Comment (andrewballantyne) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-host-app.ts:4]: Will be removed in https://github.com/openshift/console/pull/11043/files#diff-022d760da705a74cf2f3dc3b0adf914fac1387ff8a6b4363821f6fb9e8e8cbf2L1

Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-host-app.ts:4]: Per discussion in slack, I will undo my changes to these files as it is removed in: https://github.com/openshift/console/pull/11043

File: frontend/packages/console-dynamic-plugin-sdk/src/index.ts [hunk 7]
```diff
@@ -1,3 +1,9 @@
+/**
+ * @file monorepo entrypoint for the dynamic plugin SDK
```
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/index.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/index.ts [hunk 8]
```diff
@@ -1,3 +1,9 @@
...
+ *
+ * Not published to NPM
```
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/index.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-core.ts [hunk 9]
```diff
@@ -1,2 +1,8 @@
+/**
+ * @file lib-core entrypoint for the dynamic plugin SDK. Provides core APIs, types and utilities used by dynamic plugins at runtime.
```
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-core.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-internal-kubevirt.ts [hunk 10]
```diff
@@ -1 +1,7 @@
+/**
+ * @file internal-api-kubevirt entrypoint for the dynamic plugin SDK. Internal package to support KubeVirt plugin migration.
```
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-internal-kubevirt.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-core.ts [hunk 11]
```diff
@@ -1,2 +1,8 @@
...
+ *
+ * Published to NPM
```
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-core.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts (hunk 2 above)
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-internal.ts]: Done!

File: frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/src/lib-webpack.ts]: Done!

**Comment by florkbr** [comment]
While we figure out the E2E - I've built this branch locally. Here is the resulting `packages/console-dynamic-plugin-sdk` zip:
[console-dynamic-plugin-sdk.zip](https://github.com/openshift/console/files/8051608/console-dynamic-plugin-sdk.zip)

**Comment by andrewballantyne** [comment]
Waiting on https://github.com/openshift/console/pull/11044

**Comment by andrewballantyne** [comment]
/retest

Hmmm... Possible flake.

```
42) Cluster Dashboard
   43) Details Card
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
      ✖ has all fields populated (3 failures)
      • has View settings link
```

**Pull request merged**